hooks/shapediver/viewer/useInteraction.ts

```diff
@@ -0,0 +1,396 @@
+		[key: string]: string[][];
+	}>({});
+
+	let nameFilter: string[] = [];
+
+	if (settings && settings.props.nameFilter !== undefined) {
+		if (typeof settings.props.nameFilter === "string") {
+			if ((settings.props.nameFilter as string).startsWith("[") && (settings.props.nameFilter as string).endsWith("]") && (settings.props.nameFilter as string) !== "[]") {
+				try {
+					nameFilter = JSON.parse(settings.props.nameFilter) as string[];
+				} catch (e) {
+					notifications.show({
+						title: "Invalid Name Filter",
+						message: "The name filter is not a valid JSON array."
+					});
+				}
+			} else {
+				nameFilter = [settings.props.nameFilter];
+			}
+		} else if (Array.isArray(settings.props.nameFilter)) {
+			nameFilter = settings.props.nameFilter;
+		}
+
+		patternRef.current = {};
+
+		for (let i = 0; i < nameFilter.length; i++) {
+			const parts = nameFilter[i].split(".");
+			const outputName = parts[0];
+			const outputId = sessionApi.getOutputByName(outputName)[0].id;
+
+			// create a regex pattern from the other parts of the array
+			// replace all "*" with ".*"
+			const patternArray = parts.slice(1).map(part => part.replace(/\*/g, ".*"));
+
+			if (!patternRef.current[outputId]) patternRef.current[outputId] = [];
+			patternRef.current[outputId].push(patternArray);
+		}
+	} else {
+		patternRef.current = {};
+	}
+
+	const selection = isInteractionSelectionParameterSettings(settings) ? settings : undefined;
+
+	/**
+	 * Callback function to add interaction data to the nodes of the output.
+	 */
+	const callback = useCallback((node?: ITreeNode) => {
+		if (!node) return;
+
+		if(multiSelectManagerRef.current || selectManagerRef.current) {
+			const manager = multiSelectManagerRef.current || selectManagerRef.current;
+
+			node.traverse(n => {
+				const interactionData = n.data.find(d => d instanceof InteractionData) as InteractionData;
+				if (interactionData)
+					manager?.deselect(n);
+			});
+
+			if (selectedNodeNames) {
+				selectedNodeNames.forEach((nodeName, i) => {
+					const parts = nodeName.split(".");
+
+					const outputApi = node.data.find(d => d instanceof OutputApiData) as OutputApiData;
+					if (!outputApi) return;
+					if (outputApi.api.name !== parts[0]) return;
+
+					node.traverse(n => {
+						if (n.getPath().endsWith(parts.slice(1).join("."))) {
+							const interactionData = n.data.find(d => d instanceof InteractionData) as InteractionData;
+							if (interactionData) {
+								manager!.select({
+									distance: i,
+									point: vec3.create(),
+									node: n
+								});
+							}
+						}
+					});
+				});
+			}
+		}
+	}, [selectedNodeNames, multiSelectManagerRef.current, selectManagerRef.current]);
+
+	for (const outputId in sessionApi.outputs) {
+		if (!patternRef.current[outputId]) patternRef.current[outputId] = [];
+		useNodeInteractionData(sessionId, outputId, patternRef.current[outputId], { select: !!selection, hover: settings?.props.hover }, callback);
+	}
```

Comment:
@MajorMeerkatThe3rd calling [hooks in loops](https://legacy.reactjs.org/docs/hooks-rules.html) isn't allowed, let's discuss this in a call.